bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BulkMutation.java

```diff
@@ -223,136 +211,79 @@ protected void markCompletion() {
     @VisibleForTesting
     synchronized void handleResult(List<MutateRowsResponse> results) {
       mutateRowsFuture = null;
-      AtomicReference<Long> backoffTime = new AtomicReference<>();
+      if (results.size() != 1) {
+        setFailure(io.grpc.Status.INTERNAL.withDescription("Expected only a single response.")
+            .asRuntimeException());
+        return;
+      }
```

Comment:
Would you mind adding a TODO to add support for multiple MutateRowsResponses? The server doesn't currently send them, but in the future it will support sub-batching of the Mutations